### PR TITLE
macOS: handle the case of GUI without interactive session

### PR DESCRIPTION
**Pull request by Axel-Naumann**
# This Pull request:

## Checklist:

- [ ] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes #

### CODE REVIEW

File: gui/gui/src/TGClient.cxx [hunk 1]
```diff
@@ -146,8 +146,6 @@ TGClient::TGClient(const char *dpyName)
 
    // Open the connection to the display
    if ((fXfd = gVirtualX->OpenDisplay(dpyName)) < 0) {
-      Error("TGClient", "can't open display \"%s\", switching to batch mode...\n In case you run from a remote ssh session, reconnect with ssh -Y",
-            gVirtualX->DisplayName(dpyName));
```
Comment (bellenot) [inline gui/gui/src/TGClient.cxx:150]: Why do you completely remove the message?

**Review by bellenot — APPROVED** [review]
LGTM. Not sure if the error message must be removed though...

File: gui/gui/src/TGClient.cxx (hunk 1 above)
Comment (Axel-Naumann) [reply gui/gui/src/TGClient.cxx:150]: See the lengthy commit message in [`ae2f4f1` (#9314)](https://github.com/root-project/root/pull/9314/commits/ae2f4f1b203da94bd44c7933cbbebd7922408f57) - does that answer it?

Comment (bellenot) [reply gui/gui/src/TGClient.cxx:150]: Yes, sorry, I missed it...

**Comment by couet** [comment]
From Timur:

_I think the patch is fine, though I’d probably use kCGNullDirectDisplay (https://developer.apple.com/documentation/coregraphics/kcgnulldirectdisplay?language=objc) and is that 0 after all._

**Review by couet — APPROVED** [review]

**Pull request merged**